airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/cdc/PostgresCdcCtidInitializer.java

```diff
@@ -238,8 +242,17 @@ public static List<AutoCloseableIterator<AirbyteMessage>> cdcCtidIteratorsCombin
     final Supplier<AutoCloseableIterator<AirbyteMessage>> incrementalIteratorSupplier = () -> handler.getIncrementalIterators(
         propertiesManager, eventConverter, new PostgresCdcSavedInfoFetcher(stateToBeUsed), postgresCdcStateHandler);
 
+    final List<AutoCloseableIterator<AirbyteMessage>> allStreamsCompleteStatusEmitters = catalog.getStreams().stream()
+        .filter(stream -> stream.getSyncMode() == SyncMode.INCREMENTAL)
```

Comment:
This may include incremental streams that need to go through through initial snapshot.
On the other connector we're checking the equivalent of
```
.filter(stream → ! finalListOfStreamsToBeSyncedViaCtid.contains(stream))
```